### DISCUSSION TITLE
##[error]Bad credentials

### DISCUSSION THREAD

**Discussion by Schneegans** [community/community#25035] [Actions]
<p>We’re trying to use the <a href="https://github.com/svenstaro/upload-release-action" rel="nofollow noopener">svenstaro/upload-release-action</a> in <a href="https://github.com/cosmoscout/cosmoscout-vr/tree/feature/actions" rel="nofollow noopener">CosmoScout VR</a>. While it works very nicely for our workflows running on Linux, it somehow fails with this error on a Windows host.</p>
<pre><code>##[error]Bad credentials
</code></pre>
<p>Here is the error: <a href="https://github.com/cosmoscout/cosmoscout-vr/runs/234662696#step:7:8" rel="nofollow noopener">https://github.com/cosmoscout/cosmoscout-vr/runs/234662696#step:7:8</a></p>
<p>Here is the according workflow file: <a href="https://github.com/cosmoscout/cosmoscout-vr/blob/feature/actions/.github/workflows/release.yml" rel="nofollow noopener">https://github.com/cosmoscout/cosmoscout-vr/blob/feature/actions/.github/workflows/release.yml</a></p>
<p>After some testing, it seems that a significant difference between our workflow jobs are the build times: Linux takes only 30 min while Windows requires around 1.5 h. <span>If the build times are lowered artificially, the upload works just fine. </span>Can it be possible that the credentials provided by GitHub expire to quickly?</p>
<p>Related issue at the action’s repository: <a href="https://github.com/svenstaro/upload-release-action/issues/1" rel="nofollow noopener">https://github.com/svenstaro/upload-release-action/issues/1</a></p>
<p><strong>Edit:</strong> Today, after one week, I verified that this issue still persists… Is this a known bug or am I doing something wrong here?</p>

**Comment by Schneegans** [comment]
<p>This is still an issue…</p>

**Comment by Schneegans** [comment, marked as answer]
<p>While I did not get any response yet, I found an <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token#about-the-github_token-secret" rel="nofollow noopener">official documentation on this issue</a>!</p>
<p><em>The installation access token expires after 60 minutes. GitHub fetches a token for each job, before the job begins.</em></p>
<p>So as a workaround, will use a personal access token instead of <code>secrets.GITHUB_TOKEN</code>.</p>

**Comment by Trass3r** [comment]
<p>The only way to make it work seems to be using a separate job for release creation and pass the files between them via artifacts: <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/persisting-workflow-data-using-artifacts#passing-data-between-jobs-in-a-workflow" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/persisting-workflow-data-using-artifacts#passing-data-between-jobs-in-a-workflow</a>.</p>
<p>Which is very inconvenient and a lot of boilerplate!</p>

**Comment by slackhappy** [comment]
<p>Using a separate job and passing artifacts using upload/download doesn’t work, unforutnately.  The expiry timer seems to kick off when the workflow containing the jobs start,  not when the job starts.  I tried the multi-job approach and got the same bad credentials result, even though the job that used those was  configured to start only after the long-running build job ended (via needs).  Now that I’m reading the docs carefully, that’s what it says:</p>
<p><strong>Note:</strong> When a workflow run or its jobs are queued for more than one hour, the token may expire before the job starts.</p>
<p>For jobs the want to use credentials after the expiry period, the only way seems be to create and use a Personal Access Token</p>
